common/service/pldm/pldm_monitor.c

```diff
@@ -96,6 +98,136 @@ uint8_t pldm_get_sensor_reading(void *mctp_inst, uint8_t *buf, uint16_t len, uin
 	return PLDM_SUCCESS;
 }
 
+uint16_t pldm_platform_monitor_read(void *mctp_inst, mctp_ext_params ext_params,
+				    pldm_platform_monitor_commands_t cmd, uint8_t *req,
+				    uint16_t req_len, uint8_t *rbuf, uint16_t rbuf_len)
+{
+	/* return read length */
+	CHECK_NULL_ARG_WITH_RETURN(mctp_inst, 0);
+	CHECK_NULL_ARG_WITH_RETURN(req, 0);
+	CHECK_NULL_ARG_WITH_RETURN(rbuf, 0);
+
+	pldm_msg msg = { 0 };
+
+	memcpy(&msg.ext_params, &ext_params, sizeof(mctp_ext_params));
+
+	msg.hdr.pldm_type = PLDM_TYPE_PLAT_MON_CTRL;
+	msg.hdr.cmd = cmd;
+	msg.hdr.rq = 1;
+
+	msg.buf = req;
+	msg.len = req_len;
+
+	LOG_HEXDUMP_DBG(msg.buf, msg.len, "Buf: ");
+
+	return mctp_pldm_read(mctp_inst, &msg, rbuf, rbuf_len);
+}
+uint8_t pldm_encode_sensor_event_data(struct pldm_sensor_event_data *sensor_event,
```

Comment:
Do we plan to use this outside this file?

If not, we can make it static and remove the header reference.